### PR TITLE
Rename fanoutprocessor to FanOutConnector

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This implements https://github.com/open-telemetry/opentelemetry-service/issues/283

### CODE REVIEW

File: processor/fanoutconnector.go [hunk 1]
```diff
@@ -12,30 +12,30 @@
...
 
-type metricsConsumers []consumer.MetricsConsumer
+type metricsFanOutConnector []consumer.MetricsConsumer
 
-var _ processor.MetricsProcessor = (*metricsConsumers)(nil)
+var _ MetricsProcessor = (*metricsFanOutConnector)(nil)
 
 // ConsumeMetricsData exports the MetricsData to all consumers wrapped by the current one.
-func (mcs metricsConsumers) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
+func (mcs metricsFanOutConnector) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
```
Comment (pjanotti) [inline processor/fanoutconnector.go]: nit of nitness: `mcs` now doesn't match the type as before, what about `mfc`? 

While at that `mdp` on the loop also doesn't match well (seems to refer to MetricsDataProcessor), `mc` seems better.

File: processor/fanoutconnector.go [hunk 2]
```diff
@@ -45,17 +45,17 @@ func (mcs metricsConsumers) ConsumeMetricsData(ctx context.Context, md consumerd
...
 
-type traceConsumers []consumer.TraceConsumer
+type traceFanOutConnector []consumer.TraceConsumer
 
-var _ processor.TraceProcessor = (*traceConsumers)(nil)
+var _ TraceProcessor = (*traceFanOutConnector)(nil)
 
 // ConsumeTraceData exports the span data to all trace consumers wrapped by the current one.
-func (tcs traceConsumers) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+func (tcs traceFanOutConnector) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
```
Comment (pjanotti) [inline processor/fanoutconnector.go]: simmetrical nit of nitness here

File: processor/fanoutconnector_test.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ func TestTraceProcessorMultiplexing(t *testing.T) {
 		processors[i] = &mockTraceConsumer{}
 	}
 
-	tdp := NewTraceProcessor(processors)
+	tdp := NewTraceFanOutConnector(processors)
```
Comment (pjanotti) [inline processor/fanoutconnector_test.go]: not from your change but I see that `tdp` also is not a good name...

File: processor/fanoutconnector.go (hunk 1 above)
Comment (tigrannajaryan) [reply processor/fanoutconnector.go]: I'll fix.

File: processor/fanoutconnector_test.go (hunk 3 above)
Comment (tigrannajaryan) [reply processor/fanoutconnector_test.go]: I agree, fixing.

File: processor/fanoutconnector.go (hunk 1 above)
Comment (tigrannajaryan) [reply processor/fanoutconnector.go]: Fixed.

File: processor/fanoutconnector.go (hunk 2 above)
Comment (tigrannajaryan) [reply processor/fanoutconnector.go]: Fixed.

File: processor/fanoutconnector_test.go (hunk 3 above)
Comment (tigrannajaryan) [reply processor/fanoutconnector_test.go]: Fixed.

**Comment by tigrannajaryan** [comment]
Please have another look.

File: processor/fanoutconnector_test.go (hunk 3 above)
Comment (pjanotti) [reply processor/fanoutconnector_test.go]: This one is still `tdp`

Comment (tigrannajaryan) [reply processor/fanoutconnector_test.go]: Sorry, missed in the tests. Fixed now.

**Pull request merged**